x-pack/legacy/plugins/apm/public/components/app/TransactionDetails/WaterfallWithSummmary/TraceSummary.tsx

```diff
@@ -0,0 +1,255 @@
+      >
+        <>{timestamp}</>
+      </EuiToolTip>
+    </Item>
+  );
+};
+
+const SeparatorText = styled(EuiText)`
+  color: ${theme.euiColorMediumShade};
+  display: inline;
+`;
+const Separator: React.FunctionComponent = () => (
+  <>
+    {' '}
+    <SeparatorText aria-hidden={true}>|</SeparatorText>{' '}
+  </>
+);
+
+interface DurationProps {
+  duration: number;
+  totalDuration?: number;
+}
+const Duration: React.FunctionComponent<DurationProps> = ({
+  duration,
+  totalDuration
+}) => {
+  totalDuration = totalDuration || duration;
+  const isOver100 = duration > totalDuration;
+  const percentOfTrace = isOver100
+    ? '>100%'
+    : asPercent(duration, totalDuration, '');
+  const percentOfTraceText = i18n.translate(
+    'xpack.apm.transactionDetails.percentOfTrace',
+    {
+      defaultMessage: '{value} of trace',
+      values: { value: percentOfTrace }
+    }
+  );
+
+  return (
+    <Item>
+      <EuiToolTip
+        content={i18n.translate('xpack.apm.transactionDetails.durationLabel', {
+          defaultMessage: 'Duration'
+        })}
+      >
+        <>{asTime(duration)}</>
+      </EuiToolTip>{' '}
+      {isOver100 ? (
+        <EuiToolTip
+          content={i18n.translate(
+            'xpack.apm.transactionDetails.percentOfTraceLabelExplanation',
+            {
+              defaultMessage:
+                'The % of trace exceeds 100% because this transaction takes longer than the root transaction.'
+            }
+          )}
+        >
+          <>({percentOfTraceText})</>
+        </EuiToolTip>
+      ) : (
+        `(${percentOfTraceText})`
+      )}
+    </Item>
+  );
+};
+
+const MethodText = styled('span')`
+  font-weight: bold;
+`;
+
+interface HttpStatusBadgeProps {
+  status?: number;
+}
+const HttpStatusBadge: React.FunctionComponent<HttpStatusBadgeProps> = ({
+  status
+}) => {
+  let color = 'default';
+
+  if (!status) {
+    return null;
+  } else if (status < 300) {
+    color = theme.euiColorVis0;
+  } else if (status >= 300 || status < 400) {
+    color = theme.euiColorVis5;
+  } else if (status >= 400 || status < 500) {
+    color = theme.euiColorVis7;
+  } else if (status >= 500) {
+    color = theme.euiColorVis2;
+  }
+
+  return (
+    <EuiToolTip
+      content={i18n.translate('xpack.apm.transactionDetails.statusCode', {
+        defaultMessage: 'Status code'
+      })}
+    >
+      <EuiBadge color={color}>{status}</EuiBadge>
+    </EuiToolTip>
+  );
+};
+
+const HttpInfoBadge = styled(EuiBadge)`
+  max-width: 250px;
+`;
+
+interface HttpInfoProps {
+  transaction: Transaction;
+}
+const HttpInfo: React.FunctionComponent<HttpInfoProps> = ({ transaction }) => {
+  const method = (
+    idx(transaction, _ => _.http.request.method) || ''
+  ).toUpperCase();
+  const isRumAgent = isRumAgentName(transaction.agent.name);
+  const url = isRumAgent
+    ? idx(transaction, _ => _.transaction.page.url) || NOT_AVAILABLE_LABEL
+    : idx(transaction, _ => _.url.full) || NOT_AVAILABLE_LABEL;
```

Comment:
I'm not 100% but I don't think a RUM transaction will ever have `http`. We have to double check with @vigneshshanmugam or @jahtalab .

If that's the case we should remove the `transaction.page.url` from `HttpInfo` (since it will never be rendered) and instead render the RUM url as a separate use case aka:
```js
if(isRumAgent()) {
  <RumAgentUrl />
} else if (isHttp) (
 <HttpInfo />
} else if(result) {
  <TransactionResult />
}
```

Reply:
As @sqren  mentioned, We don't have `transaction.http` field at the moment for RUM agent.

Reply:
Thank @vigneshshanmugam. Do you plan to add it in the near-ish future?

Reply:
I changed this to remove the check for http which really wasn't necessary in the first place.

Reply:
Atleast for spans, we have the http info, But cant for transaction everything is grouped under page, request and response so wont be added in the nearish future.